deepchem/models/torch_model.py

```diff
@@ -0,0 +1,1167 @@
+import numpy as np
+import torch
+import torch.utils.tensorboard
+import time
+import logging
+import os
+try:
+  from collections.abc import Sequence as SequenceCollection
+except:
+  from collections import Sequence as SequenceCollection
+
+logger = logging.getLogger(__name__)
+
+from deepchem.data import Dataset, NumpyDataset
+from deepchem.metrics import Metric
+from deepchem.models.losses import Loss
+from deepchem.models.models import Model
+from deepchem.models.optimizers import Adam, Optimizer, LearningRateSchedule
+from deepchem.trans import Transformer, undo_transforms
+from deepchem.utils.evaluate import GeneratorEvaluator
+
+from typing import Any, Callable, Dict, Iterable, List, Optional, Sequence, Tuple, Union
+from deepchem.utils.typing import LossFn, OneOrMany
+
+try:
+  import wandb
+  wandb.ensure_configured()
+  if wandb.api.api_key is None:
+    _has_wandb = False
+    wandb.termwarn(
+        "W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable."
+    )
+  else:
+    _has_wandb = True
+except (ImportError, AttributeError):
+  _has_wandb = False
+
+
+def is_wandb_available():
+  return _has_wandb
+
+
+class TorchModel(Model):
```

Comment:
This might be out of the scope of this PR, but it might be really useful to add a page to `deepchem/docs` that explains how to develop custom models in DeepChem with Keras and PyTorch examples. Now that our infrastructure is maturing, we're likely to see more folks who use DeepChem for custom applications so a guide might be really useful

Reply:
Yes, definitely!  (But that's a separate PR.)